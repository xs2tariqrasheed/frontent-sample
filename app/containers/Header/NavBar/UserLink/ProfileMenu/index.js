/*
 * ProfileMenu
 *
 * ProfileMenu for the user
 *
 */

import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { BsPencil } from 'react-icons/bs';
import { GiAlliedStar } from 'react-icons/gi';
import PropTypes from 'prop-types';
import { useInjectSaga } from 'utils/injectSaga';
import * as fcl from '@onflow/fcl';
import {
  NavbarDropdown,
  NavbarDropdownContent,
  GenericLink,
  LogOut,
  GenericMenuItem,
  MedalsContainer,
} from './styled';
import { clearUserData, clearMyGolfers } from './actions';
import { loadMyMedals, myMedalsLoaded } from '../actions';
import profile from '../../../../../images/profile.png';
import saga from '../saga';

/*
Add back later when we have a profile to show.

<Link to="profile">Profile</Link>
      <div style={{ marginTop: '5px' }} />
*/

const ProfileMenu = props => {
  useInjectSaga({ key: 'profileMenu', saga });

  useEffect(() => {
    props.dispatch(loadMyMedals());
  }, []);

  return (
    <div
      style={
        {
          // backgroundColor: 'red',
        }
      }
    >
      <NavbarDropdown>
        <img
          src={profile}
          style={{ height: 30, width: 30, objectFit: 'cover', marginTop: 5 }}
        />
        {/* <GenericLink to="#" /> */}
        <NavbarDropdownContent>
          <div className="inline-block mr-2">
            <GenericMenuItem to="/profile">{props.username}</GenericMenuItem>
          </div>
          <div className="inline-block align-middle">
            <BsPencil />
          </div>
          <MedalsContainer>
            <div
              className="inline-block ml-2 align-middle"
              style={{ color: '#fcea3b' }}
            >
              <GiAlliedStar />
            </div>
            <p>
              {props.medals.medals}/{props.medals.maxmedals}
            </p>
          </MedalsContainer>
          <GenericMenuItem to="/profile">Wallet</GenericMenuItem>
          <GenericMenuItem
            onClick={() => {
              props.migrateCallback();
              window.dataLayer.push({
                event: 'migrate golfers button',
                event_source_page: 'navigation',
              });
            }}
            to="#"
          >
            Migrate to Flow
          </GenericMenuItem>
          <div style={{ marginTop: '5px' }} />
          <LogOut
            onClick={async () => {
              await fcl.currentUser().unauthenticate();
              props.dispatch(clearUserData());
              props.dispatch(clearMyGolfers());
              window.dataLayer.push({
                event: 'update user property',
                auth_status: 'not authenticated',
              });

              window.dataLayer.push({
                event: 'logged out',
                event_source_page: window.location.href,
                event_version: 'v2',
                auth_status: 'not authenticated',
              });
            }}
          >
            Logout
          </LogOut>
        </NavbarDropdownContent>
      </NavbarDropdown>
    </div>
  );
};

ProfileMenu.propTypes = {
  username: PropTypes.string,
  dispatch: PropTypes.any,
  migrateCallback: PropTypes.func,
};

function mapProps(store) {
  return {
    medals: store.headerInfo.medals,
  };
}

export default connect(mapProps)(ProfileMenu);
