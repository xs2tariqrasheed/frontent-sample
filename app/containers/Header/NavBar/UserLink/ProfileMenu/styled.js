import styled from 'styled-components';
import { Link } from 'react-router-dom';
const statsBg = 'https://www.blockletegames.com/images/DotBackground.png';

export const NavbarDropdownContent = styled.div`
  display: none;
  width: 200px;
  position: absolute;
  right: 20px;
  border-radius: 6px;
  border: solid 3px #322c8f;
  box-shadow: 0px 5px rgba(50, 44, 143, 1);
  padding: 25px;
  z-index: 1;
  background-image: linear-gradient(
      to bottom,
      rgba(255, 255, 255, 0) 1%,
      #ffffff 56%
    ),
    url(${statsBg});
  background-size: 1000px;
`;

export const NavbarDropdown = styled.div`
  position: relative;
  display: inline-block;
  // display: none;

  &:hover ${NavbarDropdownContent} {
    display: block;
  }
`;

export const MedalsContainer = styled.div`
  background-color: #313131;
  border-radius: 5px;
  margin: 4px 30px 4px 0;
  p {
    color: #ffffff;
    font-family: Industry;
    font-size: 14px;
    font-style: italic;
    padding: 2px;
    margin: 0;
    display: inline-block;
  }
`;

export const GenericLink = styled.span`
  margin-right: 20px;
  font-family: Proxima-nova;
  font-size: 16px;
  font-weight: bold;
  text-decoration: none;
  color: #322c8f;
  cursor: pointer;

  :hover {
    color: #0ac47e;
    text-decoration: none;
  }
`;

export const LogOut = styled.div`
  font-family: Proxima-nova;
  text-decoration: none;
  color: #322c8f;
  font-size: 16px;
  font-weight: bold;
  line-height: 2;
  cursor: pointer;
  display: block;

  :hover {
    color: #0ac47e;
    text-decoration: none;
  }
`;

export const MyGolfLink = styled(Link)`
  font-family: Oswald;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: 1px;
  text-decoration: none;
  color: black;
`;

export const GenericMenuItem = styled(Link)`
  font-family: Proxima-nova;
  text-decoration: none;
  color: #322c8f;
  font-size: 16px;
  font-weight: bold;
  line-height: 2;
  cursor: pointer;
  display: block;

  :hover {
    color: #0ac47e;
    text-decoration: none;
  }
`;
