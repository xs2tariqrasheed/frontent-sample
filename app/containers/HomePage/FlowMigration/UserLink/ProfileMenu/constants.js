/*
 *
 * Profile constants
 *
 */

export const CLEAR_MY_GOLFERS = 'app/UserGolfers/CLEAR_MY_GOLFERS';
export const CLEAR_USER_DATA = 'app/Login/CLEAR_USER_DATA';
