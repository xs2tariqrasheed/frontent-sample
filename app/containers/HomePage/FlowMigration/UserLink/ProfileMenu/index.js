/*
 * ProfileMenu
 *
 * ProfileMenu for the user
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as fcl from '@onflow/fcl';
import {
  NavbarDropdown,
  NavbarDropdownContent,
  GenericLink,
  LogOut,
  MyGolfLink,
  GenericMenuItem,
} from './styled';
import { clearUserData, clearMyGolfers } from './actions';

/*
Add back later when we have a profile to show.

<Link to="profile">Profile</Link>
      <div style={{ marginTop: '5px' }} />
*/
const ProfileMenu = props => {
  const link = '/clubhouse';

  return (
    <NavbarDropdown>
      <GenericLink to='#'>{props.username}</GenericLink>
      <NavbarDropdownContent>
        <MyGolfLink to={link}>My Clubhouse</MyGolfLink>
        <GenericMenuItem
          onClick={() => {
            props.migrateCallback();
          }}
          to='#'
        >
          Migrate to Flow
        </GenericMenuItem>
        <div style={{ marginTop: '5px' }} />
        <LogOut
          onClick={async () => {
            await fcl.currentUser().unauthenticate();
            props.dispatch(clearUserData());
            props.dispatch(clearMyGolfers());
          }}
        >
          Logout
        </LogOut>
      </NavbarDropdownContent>
    </NavbarDropdown>
  );
};

ProfileMenu.propTypes = {
  username: PropTypes.string,
  dispatch: PropTypes.any,
  migrateCallback: PropTypes.func,
};

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default connect(mapDispatchToProps)(ProfileMenu);
