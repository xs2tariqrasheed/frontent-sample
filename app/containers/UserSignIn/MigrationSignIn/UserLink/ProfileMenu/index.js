/*
 * ProfileMenu
 *
 * ProfileMenu for the user
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import * as fcl from '@onflow/fcl';
import {
  NavbarDropdown,
  NavbarDropdownContent,
  LogOut,
  MyGolfLink,
} from './styled';
import { clearUserData, clearMyGolfers } from './actions';
import { Wrapper } from '../styled';
// import agreeButton from '../../../../../images/buttons/buttons-desktop-green-migrate-golfers.svg';

/*
Add back later when we have a profile to show.

<Link to="profile">Profile</Link>
      <div style={{ marginTop: '5px' }} />
*/
const ProfileMenu = props => {
  const link = '/clubhouse';

  return (
    <Wrapper>
      <a
        type="submit"
        className="button"
        onClick={() => {
          props.migrateCallback();
        }}
      >
        <div className="inner-circle" />
        <div className="inner-shadow" />
        <p className="button-text">Migrate Tokens</p>
      </a>
    </Wrapper>
  );
};

ProfileMenu.propTypes = {
  username: PropTypes.string,
  dispatch: PropTypes.any,
  migrateCallback: PropTypes.func,
};

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default connect(mapDispatchToProps)(ProfileMenu);
