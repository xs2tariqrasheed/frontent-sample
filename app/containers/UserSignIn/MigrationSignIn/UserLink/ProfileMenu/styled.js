import styled from 'styled-components';
import { Link } from 'react-router-dom';
import theme from '../../../../../styles/theme';

export const NavbarDropdownContent = styled.div`
  display: none;
  position: absolute;
  right: 20px;
  background-color: ${theme.color.lightBackground};
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0, 2);
  padding: 12px 16px;
  z-index: 1;
`;

export const NavbarDropdown = styled.div`
  position: relative;
  display: inline-block;
  // display: none;

  &:hover ${NavbarDropdownContent} {
    display: block;
  }
`;

export const GenericLink = styled.span`
  margin-right: 15px;
  font-family: Oswald;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: 1px;
  color: #ffffff;
  text-decoration: none;

  :hover {
    color: ${theme.color.tin};
    text-decoration: underline;
  }
`;

export const LogOut = styled.div`
  cursor: pointer;
  font-family: Oswald;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: 1px;
  color: ${theme.color.tin};
`;

export const MyGolfLink = styled(Link)`
  font-family: Oswald;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1;
  letter-spacing: 1px;
  text-decoration: none;
  color: black;
`;

export const GenericMenuItem = styled.div`
  font-family: Roboto;
  text-decoration: none;
  color: black;
  cursor: pointer;
`;
