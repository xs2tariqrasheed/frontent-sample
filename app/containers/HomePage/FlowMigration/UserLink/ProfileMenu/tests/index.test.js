import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { BrowserRouter, browserHistory } from 'react-router-dom';
import configureStore from '../../../../../../configureStore';

import ProfileMenu from '../index';

describe('<ProfileMenu />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <BrowserRouter>
        <Provider store={store.store}>
          <IntlProvider locale="en">
            <ProfileMenu />
          </IntlProvider>
        </Provider>
      </BrowserRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
