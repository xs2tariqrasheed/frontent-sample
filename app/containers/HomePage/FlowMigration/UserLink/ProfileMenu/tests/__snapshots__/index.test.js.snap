// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMenu /> should render and match the snapshot 1`] = `
<div
  class="styled__NavbarDropdown-sc-359dlj-1 hdVRkf"
>
  <span
    class="styled__GenericLink-sc-359dlj-2 dcBtbm"
  />
  <div
    class="styled__NavbarDropdownContent-sc-359dlj-0 hLQxKZ"
  >
    <a
      class="styled__MyGolfLink-sc-359dlj-4 bXFcLe"
      href="/clubhouse"
    >
      My Clubhouse
    </a>
    <div
      class="styled__GenericMenuItem-sc-359dlj-5 kCgokK"
    >
      Migrate to Flow
    </div>
    <div
      style="margin-top: 5px;"
    />
    <div
      class="styled__LogOut-sc-359dlj-3 bJwIyn"
    >
      Logout
    </div>
  </div>
</div>
`;
